functests/utils/dynamic/dynamic.go

```diff
@@ -0,0 +1,63 @@
+package dynamic
+
+import (
+	"os"
+
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientGoDynamic "k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Client provides a struct for interacting with the openshift api
+type Client struct {
+	dynamicInterface clientGoDynamic.Interface
+	resource         map[string]clientGoDynamic.NamespaceableResourceInterface
+}
+
+// New returns a client with the given kubeconfig or env variable if empty.
+func New(kubeconfig string) *Client {
+	var config *rest.Config
+	var err error
+
+	if kubeconfig == "" {
+		kubeconfig = os.Getenv("KUBECONFIG")
+	}
+
+	if kubeconfig != "" {
+		glog.V(4).Infof("Loading kube client config from path %q", kubeconfig)
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		glog.V(4).Infof("Using in-cluster kube client config")
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		glog.Infof("Failed to init kubernetes client, please check the $KUBECONFIG environment variable")
+		return nil
+	}
+
+	client := &Client{}
+	client.dynamicInterface = clientGoDynamic.NewForConfigOrDie(config)
+	client.resource = make(map[string]clientGoDynamic.NamespaceableResourceInterface)
+
+	return client
+}
+
+// ConfigGVR is used to add a new resource interface to the dynamic client using a schema.GVR
+func (c *Client) ConfigGVR(gvrs ...schema.GroupVersionResource) {
+	for _, gvr := range gvrs {
+		c.resource[gvr.Resource] = c.dynamicInterface.Resource(gvr)
+	}
+}
+
+// Resource returns a namespaceable interface for the given resource
+func (c *Client) Resource(resourceName string) clientGoDynamic.NamespaceableResourceInterface {
+	return c.resource[resourceName]
+}
+
+// Assert returns a boolean to indicate the existence of an interface for the given resource name
+func (c *Client) Assert(resourceName string) bool {
```

Comment:
Note that this only checks that you have added something to the dict.
Also `c.dynamicInterface.Resource(gvr)` will not raise an error when the resource does not exist (I think it does not even return nil, but a client for a non existent resource).
Maybe in the assert you can check if the resource actually exists by trying to list it:


```
 func (c *Client) Assert(resourceName string) bool {
	clientForResource, ok := c.resource[resourceName]
       	crds, err := clientForResource.List(metav1.ListOptions{})
        return err == nil
}
```